cmd/podman/cp.go

```diff
@@ -0,0 +1,366 @@
+	ctr, err := runtime.LookupContainer(name)
+	if err != nil {
+		return errors.Wrapf(err, "error looking up container %q", name)
+	}
+	mountPoint, err := ctr.Mount()
+	if err != nil {
+		return errors.Wrapf(err, "error mounting container %q", ctr.ID())
+	}
+	user, err := getUser(mountPoint, ctr.User())
+	if err != nil {
+		return err
+	}
+	idmappingopts, err := ctr.IDMappings()
+	if err != nil {
+		return errors.Wrapf(err, "error getting IDMappingOptions")
+	}
+	containerOwner := idtools.IDPair{UID: int(user.UID), GID: int(user.GID)}
+	hostUID, hostGID, err := getHostIDs(idmappingopts.UIDMap, idmappingopts.GIDMap, int(user.UID), int(user.GID))
+	if err != nil {
+		return err
+	}
+
+	hostOwner := idtools.IDPair{UID: int(hostUID), GID: int(hostGID)}
+
+	var dest string
+	if isFromHostToContainer {
+		dest = mountPoint
+		if destination != "" && filepath.IsAbs(destination) {
+			dest = filepath.Join(dest, destination)
+		} else {
+			if err = idtools.MkdirAllAndChownNew(filepath.Join(dest, ctr.WorkingDir()), 0755, hostOwner); err != nil {
+				return errors.Wrapf(err, "error creating directory %q", filepath.Join(dest, ctr.WorkingDir()))
+			}
+			dest = filepath.Join(dest, ctr.WorkingDir(), destination)
+		}
+	} else {
+		dest = destination
+		if destination == "" || !filepath.IsAbs(destination) {
+
+			dir, err := os.Getwd()
+			if err != nil {
+				return errors.Wrapf(err, "err getting current working directory")
+			}
+			if err = idtools.MkdirAllAndChownNew(filepath.Join(dir, dest), 0755, hostOwner); err != nil {
+				return errors.Wrapf(err, "error creating directory %q", filepath.Join(dir, dest))
+			}
+			dest = filepath.Join(dir, dest)
+		}
+
+	}
+
+	if len(destination) > 0 && destination[len(destination)-1] == os.PathSeparator {
+		if err = idtools.MkdirAllAndChownNew(dest, 0755, hostOwner); err != nil {
+			return errors.Wrapf(err, "error creating directory %q", dest)
+		}
+	}
+
+	// Make sure the destination's parent directory is usable.
+	if destpfi, err2 := os.Stat(filepath.Dir(dest)); err2 == nil && !destpfi.IsDir() {
+		return errors.Errorf("%q already exists, but is not a subdirectory)", filepath.Dir(dest))
+	}
+	// Now look at the destination itself.
+	destfi, err := os.Stat(dest)
+	if err != nil {
+		if !os.IsNotExist(err) {
+			return errors.Wrapf(err, "couldn't determine what %q is", dest)
+		}
+		destfi = nil
+	}
+
+	copyFileWithTar := copyFileWithTar(idmappingopts, &containerOwner, digest.Canonical.Digester().Hash())
+	copyWithTar := copyWithTar(idmappingopts, &containerOwner, digest.Canonical.Digester().Hash())
+	untarPath := untarPath(idmappingopts, nil, digest.Canonical.Digester().Hash())
+
+	if !isFromHostToContainer {
+		src := mountPoint
+		if srcPath != "" && filepath.IsAbs(srcPath) {
+			srcPath = filepath.Join(src, srcPath)
+		} else {
+			srcPath = filepath.Join(src, ctr.WorkingDir(), srcPath)
+		}
+	}
+
+	glob, err := filepath.Glob(srcPath)
```

Comment:
I'm pretty sure this is wrong - per https://docs.docker.com/engine/reference/commandline/cp/ `docker cp` does not support globs. I think we should duplicate the functionality initially, we can think about expanding on it more.